### PR TITLE
Integrate norm helper

### PR DESCRIPTION
**Pull request by MARD1NO**
### PR types
New features

### PR changes
OPs

### Description
Add NormHelper，support layernorm / rmsnorm and biasadd_residual fusion. 
layernorm support residual_alpha, which is used in ChatGLM model: 
https://huggingface.co/THUDM/chatglm-6b/blob/main/modeling_chatglm.py#L650-L651

Fix https://github.com/PaddlePaddle/Paddle/pull/54998 review comment

Pcard-72603

### CODE REVIEW

File: paddle/phi/kernels/gpu/norm_helper_kernel.cu [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+    // Do norm.
+    norm_helper.Norm(x, norm_weight, norm_bias, mean, variance, out);
+  }
+
+#endif
+}
+
+}  // namespace phi
+
+#ifndef PADDLE_WITH_HIP
```
Comment (vivienfanghuagood) [inline paddle/phi/kernels/gpu/norm_helper_kernel.cu:232]: Why is the kernel associated with HIP？

File: paddle/phi/kernels/gpu/norm_helper_kernel.cu [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+                                       norm_weight_data,
+                                       norm_bias_data,
+                                       output_data,
+                                       mean_data,
+                                       var_data);
+    } else if (norm_type_ == "rmsnorm") {
+      // For rmsnorm, it use Input's type weight and bias.
+      // Currently, it only used in inference, so we do not save intermediate
+      // result for backward.
+      const T* norm_weight_data = norm_weight.data<T>();
```
Comment (vivienfanghuagood) [inline paddle/phi/kernels/gpu/norm_helper_kernel.cu:144]: Why suppose RMSNorm has the same type with input, but LayerNorm is not？

File: paddle/phi/api/yaml/ops.yaml [hunk 3]
```diff
@@ -1786,6 +1786,16 @@
     func : nonzero
     data_type: condition
 
+- op : norm_helper
+  args : (Tensor x, Tensor residual, Tensor bias, Tensor norm_weight, Tensor norm_bias, float epsilon, float residual_alpha, str norm_type, int begin_norm_axis)
```
Comment (vivienfanghuagood) [inline paddle/phi/api/yaml/ops.yaml:1790]: I think more descriptions about the `norm_type`, `begin_norm_axis`, `residual_alpha`  should be added.

File: python/paddle/incubate/nn/functional/rms_norm.py [hunk 4]
```diff
@@ -50,9 +50,15 @@ def rms_norm(x, weight, bias, epsilon, begin_norm_axis):
 
     helper = LayerHelper('rms_norm', **locals())
     out = helper.create_variable_for_type_inference(dtype=x.dtype)
```
Comment (zyfncg) [inline python/paddle/incubate/nn/functional/rms_norm.py:52]: norm_helper有对应的 python api 吗？

Comment (MARD1NO) [reply python/paddle/incubate/nn/functional/rms_norm.py:52]: 这个接口不成熟，暂时不暴露。后续是用在 Inference优化的LLM模型组网内部

Comment (zyfncg) [reply python/paddle/incubate/nn/functional/rms_norm.py:52]: 是只在inference场景下使用吗？

File: paddle/phi/api/yaml/ops.yaml [hunk 5]
```diff
@@ -1786,6 +1786,16 @@
     func : nonzero
     data_type: condition
 
+- op : norm_helper
```
Comment (qingqing01) [inline paddle/phi/api/yaml/ops.yaml:1789]: helper没必要暴露API，放在yaml里吧

**Pull request closed without merging**